In [ ]:
import pandas as pd
import numpy as np
import os
import sys

In [ ]:
# setting a variable for the raw file
url = 'https://raw.githubusercontent.com/fedhere/PUI2018_fb55/master/Lab4_fb55/times.txt'     

## Formulation of our Hypothesis: 
### Does the new bus route improve commuting time?
#### Null Hypothesis
H0: Communiting time is on average longer or the same with the new bus route: Tnew >= Told

#### Alternative Hypothesis
Ha: Commuting time is shorter with the new bus route: Tnew < Told

#### Significance level as: $\alpha=0.05$

In [4]:
# reading the file as a csv with a dataframe
bus = pd.read_csv(url, sep=",", names = ["Commute Time"])

In [5]:
# previewing the first five rows in the dataframe
bus.head()

Commute Time
0     31.622239
1     32.821376
2     30.229101
3     31.413766
4     39.010550

In [9]:
# Calculating mean and standard deviation
x = bus['Commute Time'].mean()
y = bus['Commute Time'].std()
N = len(bus)
print ("Mean:", x, "\nStandard Deviation:", y)

Mean: 34.4661616883 
Standard Deviation: 7.13728013375


In [10]:
pmean = 36
pstd = 6

In [12]:
# Calculating the Z score 
a = pmean - x 
b = pstd / np.sqrt(N)
zscore= a/b
print(zscore)

2.55639718617


### The Z-test score is 2.556. Being higher than our significance threshold of 0.05, we can reject the null hypothesis. Thus, the new bus route improves commuting time.